libraries/engage/locations/providers/FulfillmentProvider.jsx

```diff
@@ -0,0 +1,335 @@
+   */
+  async function sendReservation(values: ReservationFormValues) {
+    // Store the user's form in the user data.
+    storeFormInput(values);
+
+    try {
+      const response = await submitReservation(values, product);
+
+      if (response.errors && response.errors.length > 0) {
+        setStage(STAGE_RESPONSE_ERROR);
+        setOrderNumbers(null);
+        setErrors(response.errors);
+      }
+
+      setStage(STAGE_RESPONSE_SUCCESS);
+      setOrderNumbers(response.orderNumbers);
+      setErrors(null);
+    } catch (error) {
+      logger.error(error);
+      setStage(STAGE_RESPONSE_ERROR);
+      setOrderNumbers(null);
+      setErrors([error.message]);
+    }
+  }
+
+  /**
+   * Handles multiline reservation.
+   * @param {Object} location The selected location.
+   */
+  function handleMultilineReservation(location: Location) {
+    if (product === null || location.code === null) {
+      return;
+    }
+
+    const fulfillment = {
+      method: 'ROPIS', // TODO: make this dynamic.
+      location: {
+        code: location.code,
+        name: location.name || '',
+      },
+    };
+
+    if (isChangeFulfillment && cartItem) {
+      updateProductsInCart([{
+        quantity: 1,
+        cartItemId: cartItem.id,
+        fulfillment,
+      }]);
+    } else {
+      addProductsToCart([{
+        productId: product.id,
+        quantity: 1,
+        fulfillment,
+      }]);
+    }
+
+    handleClose(location, product.id);
+  }
+
+  /**
+   * Handles quick reservation.
+   */
+  function handleQuickReservation() {
+    setStage(STAGE_RESERVE_FORM);
+  }
+
+  /**
+   * Handles the selection of a store location from the sheet.
+   * @param {Object} location The selected location.
+   */
+  function handleSelectLocation(location: Location) {
+    selectLocation({
+      code: location.code,
+      name: location.name,
+    });
+
+    if (changeOnly) {
+      handleClose(location, product && product.id);
+      return;
+    }
+
+    // No fulfillment path selected yet.
+    if (fulfillmentPath === null && fulfillmentPaths.length > 1) {
+      FulfillmentPathSelector.open((method: FulfillmentPath) => {
+        if (method === FULFILLMENT_PATH_QUICK_RESERVE) {
+          handleQuickReservation();
+        }
+
+        if (method === FULFILLMENT_PATH_MULTI_LINE_RESERVE) {
+          handleMultilineReservation(location);
+        }
+      });
+      return;
+    }
+
+    if (
+      fulfillmentPath === FULFILLMENT_PATH_MULTI_LINE_RESERVE
+      && fulfillmentPaths.includes(FULFILLMENT_PATH_MULTI_LINE_RESERVE)
+    ) {
+      handleMultilineReservation(location);
+      return;
+    }
+
+    if (fulfillmentPath === FULFILLMENT_PATH_QUICK_RESERVE) {
+      handleQuickReservation();
+    }
+  }
+
+  /**
+   * @param {string} method The selected fulfillment method.
+   * @param {Object} item The cart item to change.
+   */
+  function handleChangeFulfillmentMethod(method, item) {
+    setIsChangeFulfillment(true);
```

Comment:
Not sure where bug lies. When you cancel selecting a path on PDP, switching product variant opens selecting a Path Sheet every time you change variant

Video: https://jira.shopgate.guru/secure/attachment/58553/record-09.17.17%5B11.03.2020%5D.mp4

Reply:
Seems to be fixed after solving the context issues.

Reply:
Nope. Still reproducible